posts/no-chatgpt-is-not-going-to-take-away-your-progarmming-job.md

```diff
@@ -0,0 +1,256 @@
+range_entry.grid(row=1, column=1)
+range_entry.insert(0, '100000000')
+
+# Create the start button
+start_button = tk.Button(root, text='Start', command=start_benchmark)
+start_button.grid(row=2, column=0, columnspan=2)
+
+# Create the output labels
+total_sum_label = tk.Label(root, text='Total sum: ')
+total_sum_label.grid(row=3, column=0, columnspan=2)
+
+time_taken_label = tk.Label(root, text='Time taken: ')
+time_taken_label.grid(row=4, column=0, columnspan=2)
+
+root.mainloop()
+```
+
+If you copy and past this code (I do not recommend) and run it, it will start a small gui. Mine only has a start button, the text fields are there, but you can't see them. Further more, pressing "start", causes the program to spawn multiple versions of it's self and gives my MAC the rainbow spinning wheel of death. I was able to capture a screen shot of it with activity monitor open before I killed.
+
+![Crashing Tkinter Program](../images/2023/04/tkinter-throwup.png)
+
+While this example is not actual dangerous, it is clear to me that ChatGPT has the capability to generate "system crashing" code. As always, and this goes for places like stack overflow as well, do not blindly run code you found on the internet.
+
+## The Good Stuff
+
+Okay, now for a positive note. Throughout this whole process, which I thoroughly enjoyed, I actually learned something new about python that I didn't know before. That's rigth, ChatGPT taught me something. It came in the form of a python function that I had never seen before. If you look back at the first code that ChatGPT generated for me, there is this little function:
+
+```python
+# Spawn the processes and wait for them to finish
+with multiprocessing.Pool(processes=num_processes) as pool:
+    results = pool.starmap(calc_sum, chunks)
+```
+
+This is the first time I had seen the function `starmap`. I took a quick detour to the [official python docs](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.pool.Pool.starmap) and read all about it. Obviously I could have come across this function or this piece of code in the wild or learned it by reading the docs more thoroughly, but my point is that, given this generated output from ChatGPT, I learned something. Period. The End.  
+
+# Conclusion
+
+I don't believe that generative large language models such as ChatGPT are poised to take our hard won programming jobs. There is still just to much surrounding knowledge, adjacent knowledge, that can't fit into a prompt, or that doesn't make sense for ChatGPT to have access to.  For instance, an organizations entire source code does not need to be handed over to ChatGPT, but in some cases, that is what it would take to have the given context. So rest assured, we are safe! For now... Who knows what the future holds, in the meantime, I am going to make the most out of the situation and use ChatGPT as a learning tool, to discover new functions and paradigms I may have missed, or would not otherwise know about.
```

Comment:
So rest assured, we are safe! For now...

Should probably be "So rest assured, we are safe... for now. Who knows what the future holds. In the meantime,"